### DISCUSSION TITLE
Class controllers

### DISCUSSION THREAD

**Discussion by bellaabdelouahab** [ISIL-ESTE/Student-Workflow-Organizer#138] [Q&A] [answered]
how about if we use class controllers instead of just sample methods, 
i heard that it can facilitate the swagger api generation !
what do you think ?
@muttaqin1 @Khalid1G

**Comment by ghost** [comment]
> how about if we use class controllers instead of just sample methods, 
> i heard that it can facilitate the swagger api generation !
> what do you think ?
> @muttaqin1 @Khalid1G 

Yes, Class-based controllers provide a more structured way to define routes and request handlers. We can inject dependencies into controller classes. Since we are using typescript we can use decorators to add extra functionality to our class methods.
@bellaabdelouahab

**Comment by bellaabdelouahab** [comment]
@muttaqin1 I found a nice package to help us with this 
it's called tsoa (type script open api) it uses controllers and decorators  to generate the swagger docs

**Reply by ghost** [reply, marked as answer]
> @muttaqin1 I found a nice package to help us with this 
> it's called tsoa (type script open api) it uses controllers and decorators  to generate the swagger docs

It's very interesting. Lets implement this.
@bellaabdelouahab
